### DISCUSSION TITLE
ssh into the guest

### DISCUSSION THREAD

**Discussion by ItsNovarion** [cockpit-project/cockpit-machines#1907] [Q&A] [answered]
hey there,

i am still relatively new to the cockpit and qemu things.

i have created an ubuntu vm and wanted to access it via ssh, but this is not possible because the vm is not in the local subnet.

i have already thought about port forwarding, but i have not yet managed to forward port 22 to the host port 2022.

can somebody please help me?

**Comment by jelly** [comment, marked as answer]
That is correct, the default libvirtd network does not support ssh'ing into from your own VM. Cockpit itself cannot setup port forwarding as we lack any implementation for it. So you can either do this with virt-manager or edit the xml or switch to host mode networking.

**Reply by ItsNovarion** [reply]
Hey there...

So if i wanted to access the vms via SSH oder RDP cockpit machines cant do this?

Sorry but im really new to cockpit
